src/ObjectStore/v1/Models/Object.php

```diff
@@ -96,11 +96,14 @@ public function retrieve()
      * distinct from fetching its metadata (a `HEAD` request). The body of an object is not fetched by default to
      * improve performance when handling large objects.
      *
+     * @param array $options {@see \OpenStack\ObjectStore\v1\Api::getObject} 
+     *
      * @return StreamInterface
      */
-    public function download(): StreamInterface
+    public function download($options=array()): StreamInterface
     {
-        $response = $this->executeWithState($this->api->getObject());
+        $obj = $this->api->getObject();
+        $response = $this->execute($obj, $options + $this->getAttrs(array_keys($obj['params'])));
```

Comment:
@ekasitk I see your point because `containerName` does not exist in `$data`.
To make it consistent

```php
$data['container$options]
$response = $this->execute($this->api->getObject(), $data);
```